Userland/Libraries/LibGUI/VimEditingEngine.cpp

```diff
@@ -1148,7 +1159,14 @@ void VimEditingEngine::clear_visual_mode_data()
     if (m_editor->has_selection()) {
         m_editor->selection()->clear();
         m_editor->did_update_selection();
+
+        auto cursor = m_editor->cursor();
+        if (cursor.column() >= m_editor->current_line().length()) {
+            cursor.set_column(m_editor->current_line().length() - 1);
+            m_editor->set_cursor(cursor);
+        }
```

Comment:
Entering Vim emulation mode for the first time needs this too, so might make sense to put this in a separate function so both constructor and `clear_visual_mode_data` can use it.

Reply:
See https://github.com/SerenityOS/serenity/pull/6684/files#diff-c13c2f0b49898f9e679fc5f13025ba8d27d9607fb40760810b906da16cbc3785R1157-R1165